src/Symfony/Component/HttpFoundation/JsonResponse.php

```diff
@@ -27,9 +27,9 @@ class JsonResponse extends Response
     protected $data;
     protected $callback;
 
-    // Encode <, >, ', &, and " characters in the JSON, making it also safe to be embedded into HTML.
-    // 15 === JSON_HEX_TAG | JSON_HEX_APOS | JSON_HEX_AMP | JSON_HEX_QUOT
-    public const DEFAULT_ENCODING_OPTIONS = 15;
+    // Encode <, >, ', &, and " characters in the JSON, making it also safe to be embedded into HTML, represents floats.
+    // 1039 === JSON_HEX_TAG | JSON_HEX_APOS | JSON_HEX_AMP | JSON_HEX_QUOT | JSON_PRESERVE_ZERO_FRACTION
+    public const DEFAULT_ENCODING_OPTIONS = 1039;
```

Comment:
This change is a hard BC break, why should we change that? There is a way to change the encoding options to allow your application to have different ones than the default one.

Reply:
you mean `setEncodingOptions` method?

Reply:
Yes, exactly that one, I think this will work for this case:
```
$response->setEncodingOptions(JsonResponse::DEFAULT_ENCODING_OPTIONS | \JSON_PRESERVE_ZERO_FRACTION);
```

Reply:
yes, it is how it works now and I thought it would be the right choice for default options, but it seems it is too late to change this 😔